### PR TITLE
feat: add demo project for the ease of use

### PR DESCRIPTION
**Pull request by thilllon**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [ ] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [x] Documentation content changes
-   [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Demo project is added and it can be run by `yarn run demo`. This library's, so called, `how-to` guidance are commented.
I hope that first comers will get helped by this demo.

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
         "registry": "https://registry.npmjs.org"
     },
     "scripts": {
+        "demo": "yarn install; yarn build; cd demo; yarn install; yarn start:dev",
```
Comment (thilllon) [inline package.json:43]: add all-in-one command

File: jest.config.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ const config: Config = {
             },
         ],
     },
+    testPathIgnorePatterns: ['<rootDir>/dist/', '<rootDir>/demo/', '<rootDir>/node_modules/'],
```
Comment (thilllon) [inline jest.config.ts:21]: exclude demo and build output directory from test target

File: .github/workflows/ci.yml
Comment (thilllon) [inline .github/workflows/ci.yml:1]: clean those comments up

File: jest.config.ts [hunk 3]
```diff
@@ -18,13 +18,14 @@ const config: Config = {
...
     testTimeout: 120_000,
     testEnvironment: 'node',
     verbose: true,
     detectLeaks: false,
     detectOpenHandles: true,
     collectCoverage: true,
-    collectCoverageFrom: ['**/*.ts', '!**/*.d.ts'],
+    collectCoverageFrom: ['**/*.ts', '!**/*.d.ts','!demo/**/*','!dist/**/*'],
```
Comment (thilllon) [inline jest.config.ts:28]: image

**Comment by jiho-kr** [comment]
If you need a general guide, the readme should complement it.

**Pull request closed without merging**